dxr/build.py

```diff
@@ -447,7 +447,7 @@ def index_file(tree, tree_indexers, path, es, index):
     # will not accept the path, so just move on.
     rel_path = relpath(path, tree.source_folder)
     try:
-        rel_path = rel_path.decode('utf-8')
+        rel_path = unicode(rel_path, 'utf-8')
```

Comment:
We know rel_path to be a str, so wouldn't it be simpler to call rel_path.decode('utf-8')? `unicode()` brings in the complexity of other exception types: the docs say it will throw ValueError if decoding fails (and it's in strict error-handling mode).

But actually, I think we should move Unicode decoding into the config layer so all paths are unicode everywhere in the system. Then we wouldn't have to deal with it here. What do you think? If we do go that way, we need to decide whether it needs (or is worth) a format version bump. It may not need one, since everything comes back from ES as unicode IIRC.